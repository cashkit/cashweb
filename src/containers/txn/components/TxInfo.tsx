import React from 'react';


export interface TxInfoProps {
    version: number,
    lockTime: number,
    size: number,
    timestamp: number,
    confirmations: number,
    blockHeight: number,
}

export const TxInfo = ({
    blockHeight,
    confirmations,
    lockTime,
    size,
    timestamp,
    version, }
    : TxInfoProps) => {
  return(
    <>
      <div className="tile is-ancestor">
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Version</p>
            <div className="content">{version}</div>
          </article>
        </div>
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Block Height</p>
            <div className="content">{blockHeight}</div>
          </article>
        </div>
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Timestamp</p>
            <div className="content">{timestamp}</div>
          </article>
        </div>
      </div>
      <div className="tile is-ancestor">
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">LockTime</p>
            <div className="content">{lockTime}</div>
          </article>
        </div>
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Confirmations</p>
            <div className="content">{confirmations}</div>
          </article>
        </div>
  
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Size</p>
            <div className="content">{size}</div>
          </article>
        </div>
      </div>
      </>
    )
  }