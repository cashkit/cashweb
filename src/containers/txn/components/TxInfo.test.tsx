import React from 'react';
import renderer from 'react-test-renderer';

import { TxInfo, TxInfoProps } from './TxInfo';

const props: TxInfoProps = {
    version: 0,
    lockTime: 0,
    size: 0,
    timestamp: 0,
    confirmations: 0,
    blockHeight: 0,
}

it('<TxInfo/> renders correctly', () => {
  const tree = renderer
    .create(<TxInfo {...props}></TxInfo>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});