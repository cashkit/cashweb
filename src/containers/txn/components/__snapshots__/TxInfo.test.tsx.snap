// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TxInfo/> renders correctly 1`] = `
Array [
  <div
    className="tile is-ancestor"
  >
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Version
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Block Height
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Timestamp
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
  </div>,
  <div
    className="tile is-ancestor"
  >
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          LockTime
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Confirmations
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Size
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
  </div>,
]
`;
